js/msp.js

```diff
@@ -109,14 +109,18 @@ var MSP = {
                     }
                     break;
                 case 9:
-                    if (this.message_checksum == data[i]) {
-                        // message received, store dataview
-                        this.dataView = new DataView(this.message_buffer, 0, this.message_length_expected);
-                    } else {
+                    if (this.message_checksum != data[i]) {
                         console.log('code: ' + this.code + ' - crc failed');
-                        this.dataView = null;
                         this.packet_error++;
+                        
+                        // if this is a dataflash read packet then use
+                        // the compressionType byte as a flag to indicate
+                        // that a crc error occurred and the packet should retry
+                        if (this.code == MSPCodes.MSP_DATAFLASH_READ) {
+                            this.message_buffer_uint8_view[6] = 255;
```

Comment:
Well I said in the issue thread that it was a messy fix… :)  I'm just learning the configurator code and brushing off my extremely rusty javascript skills so I didn't want to mess with too much and the goal was to suggest a solution - not necessarily in the best coding style.  Since the MSP message decoding is unified and just the data block is passed to the callback I didn't see how to do it in a way that wouldn't affect other messages.